src/npc.cpp

```diff
@@ -1110,6 +1112,14 @@ NpcEventsHandler::NpcEventsHandler(const std::string& file, Npc* npc) :
 	}
 }
 
+NpcEventsHandler::~NpcEventsHandler()
+{
+	std::array<int32_t, 7> eventArray = {creatureSayEvent, creatureDisappearEvent, creatureAppearEvent, creatureMoveEvent, playerCloseChannelEvent, playerEndTradeEvent, thinkEvent};
```

Comment:
static?

Reply:
@nekiro a static in the *destructor* means it will always try to delete the events of the first NpcEventsHandler to ever be deleted.

No need for the temporary though
```cpp
for (int32_t eventId : {creatureSayEvent, creatureDisappearEvent, creatureAppearEvent, creatureMoveEvent, playerCloseChannelEvent, playerEndTradeEvent, thinkEvent}) {
```